### PR TITLE
Update error message for apply validation

### PR DESCRIPTION
**Pull request by chrisst**
Add a hint that the validation failure has occurred at the root of the resource
schema to the error message. This is because the root resource has an empty
path when being validated and the path is being relied upon to provide context
into the error message.

In practice when the ID of a resource gets cleared accidentally due to a bug during Create the entire `actual` schema is null. The user facing error message when this happens is:
>  Error: Provider produced inconsistent result after apply
> 
> When applying changes to google_new_service.default, provider
“google-beta” produced an unexpected new value for was present, but now
absent.

This PR would change the message to:
> Error: Provider produced inconsistent result after apply
> 
> When applying changes to google_new_service.default, provider
"google-beta" produced an unexpected new value: Root resource was present, but
now absent.

### CODE REVIEW

File: plans/objchange/compatible.go [hunk 1]
```diff
@@ -28,12 +28,17 @@ func AssertObjectCompatible(schema *configschema.Block, planned, actual cty.Valu
 
 func assertObjectCompatible(schema *configschema.Block, planned, actual cty.Value, path cty.Path) []error {
 	var errs []error
+	var atRoot string
+	if len(path) == 0 {
```
[the remark this answers is not in the document]
Comment (chrisst) [reply plans/objchange/compatible.go:32]: On line 54 of this PR, if path is not zero it's used to print out the path of the offending field. The issue was that if path was empty, eg it was a root field, the error message that got generated from "append(errs, path.NewErrorf("inconsistent values for sensitive attribute"))" generates a message that doesn't make sense - see the message in the PR description. So IIRC the only situation that wasn't already handled well was if the path was zero.

**Comment by chrisst** [comment]
This PR was meant to assist provider developers in finding a bad configuration for a resource they are developing. There isn't an issue because it's hopefully an error situation that shouldn't make it to a shipped version of a resource. I just had to dig into code a bit to figure out what the real error was and thought I'd leave better bread crumbs for the next person.
It's been a year so I don't remember exactly how to repro but I think changing a resource's create method so clear the id but return without error will throw this.

**Comment by mildwonkey** [comment]
Thank you very much for the explanation @chrisst , now I follow you. 

If you'd like (as long as I have the ability to push to your fork), I can resolve the conflict for you, otherwise you can do it yourself and I will approve this PR - I'll leave that up to you but I'm happy to take over since you've waited so long already.

**Comment by chrisst** [comment]
conflict resolved, thanks!

**Pull request merged**